warehouse/ingest-core/src/main/java/datawave/ingest/data/config/ingest/WhindexFieldIngestHelper.java

```diff
@@ -0,0 +1,227 @@
+     * from the <code>sourceField</code> into the new <code>destField</code>, optionally deleting the source.
+     *
+     * @param eventMap
+     *            the original multimap of field -> values
+     * @return a new Multimap reflecting added Whindex entries and any removed sources
+     */
+    @Override
+    public Multimap<String,NormalizedContentInterface> processWhindexFields(Multimap<String,NormalizedContentInterface> eventMap) {
+        if (valueFieldsToWhindexConfigs.isEmpty()) {
+            return eventMap;
+        }
+
+        Multimap<String,NormalizedContentInterface> result = HashMultimap.create(eventMap);
+        Multimap<String,NormalizedContentInterface> whindices = HashMultimap.create();
+        Set<String> toRemove = new HashSet<>();
+
+        // Determine which configs apply
+        List<WhindexConfig> configs = valueFieldsToWhindexConfigs.entries().stream().map(Map.Entry::getValue)
+                        .filter(cfg -> result.containsKey(cfg.getValueField()) && result.containsKey(cfg.getSourceField())).collect(Collectors.toList());
+
+        for (WhindexConfig cfg : configs) {
+            // Check if any existing values match the config's trigger list
+            boolean match = result.get(cfg.getValueField()).stream()
+                            .anyMatch(nci -> cfg.getValues().contains(nci.getEventFieldValue()) || cfg.getValues().contains(nci.getIndexedFieldValue()));
+            if (!match)
+                continue;
+
+            // Copy source entries into new dest field
+            Collection<NormalizedContentInterface> sources = result.get(cfg.getSourceField());
+            for (NormalizedContentInterface nci : sources) {
+                whindices.put(cfg.getDestField(), new NormalizedFieldAndValue(nci));
+            }
+
+            if (cfg.isOverloaded()) {
+                toRemove.add(cfg.getSourceField());
+            }
+        }
+
+        // Remove any overloaded source fields and add in new whindex entries
+        toRemove.forEach(result::removeAll);
+        result.putAll(whindices);
+
+        return result;
+    }
+
+    /**
+     * Checks if the given field name is one of the Whindex-generated destination fields.
+     *
+     * @param field
+     *            the field name to check
+     * @return true if this helper can generate values for that field
+     */
+    @Override
+    public boolean isWhindexField(String field) {
+        return destinationFields.contains(field);
+    }
+
+    /**
+     * Retrieves an immutable view of the mapping from trigger fields to configs.
+     *
+     * @return a Guava ImmutableMultimap of valueField -> WhindexConfig
+     */
+    @Override
+    public ImmutableMultimap<String,WhindexConfig> getValueFieldsToWhindexConfigs() {
+        return ImmutableMultimap.copyOf(valueFieldsToWhindexConfigs);
```

Comment:
Rather than returning an immutable copy here, let's make valueFieldsToWhindexConfig and destinationFields always an immutable lists type. If you make them non-final, then you can replace them every time setup() is called, and we can avoid copying more often then we need to.